presto-geospatial-toolkit/src/main/java/com/facebook/presto/geospatial/GeometryUtils.java

```diff
@@ -104,40 +103,42 @@ public static Envelope deserializeEnvelope(Slice shape)
         }
         BasicSliceInput input = shape.getInput();
 
-        input.readInt(); // spatial reference ID
-
         Envelope overallEnvelope = null;
-        while (input.available() > 0) {
-            int length = input.readInt();
-            ByteBuffer buffer = input.readSlice(length).toByteBuffer().order(LITTLE_ENDIAN);
-            int type = buffer.getInt();
-            Envelope envelope = null;
-            if (type == POINT_TYPE) {    // point
-                double x = buffer.getDouble();
-                double y = buffer.getDouble();
-                if (!isEsriNaN(x)) {
-                    Verify.verify(!isEsriNaN(y));
-                    envelope = new Envelope(x, y, x, y);
-                }
-            }
-            else {
-                double xMin = buffer.getDouble();
-                double yMin = buffer.getDouble();
-                double xMax = buffer.getDouble();
-                double yMax = buffer.getDouble();
-                if (!isEsriNaN(xMin)) {
-                    Verify.verify(!isEsriNaN(xMax));
-                    Verify.verify(!isEsriNaN(yMin));
-                    Verify.verify(!isEsriNaN(yMax));
-                    envelope = new Envelope(xMin, yMin, xMax, yMax);
-                }
-            }
-            if (envelope != null) {
-                if (overallEnvelope == null) {
-                    overallEnvelope = envelope;
+        if (input.available() > 0) {
+            byte code = input.readByte();
+            boolean isGeometryCollection = code == GeometryTypeName.GEOMETRY_COLLECTION.ordinal();
```

Comment:
Using an enum ordinal in this manner is fragile, as changing the order of the constants in the source code will change the ordinal. Instead, use an explicitly assigned value:
```java
enum Fruit
{
    APPLE(1),
    ORANGE(2);

    private final int id;

    Fruit(int id)
    {
        this.id = id;
    }

    public int id()
    {
        return id;
    }
}

Reply:
This is the approach I used in the beginning. @mbasmanova said it's ok to use ordinal() because it's ephemeral.

Reply:
Assuming it is ephemeral is bad because

* In the future, Presto nodes might not all be the same version.
* This code is part of `presto-geospatial-toolkit`, which is generic library code. We are making assumptions about how the library will be used.